lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -145,6 +146,48 @@ class LayoutPropagation {
   FuncOp funcOp;
 };
 
+class LayoutRematerialization {
+public:
+  LayoutRematerialization(FuncOp F) : funcOp(F) {}
+  // Map the original value to the remat'ed one.
+  void addRematValue(Value old, Attribute encoding, Value newV);
+  bool hasRematValue(Value value, Attribute encoding) {
+    return rematMapping.contains({value, encoding});
+  }
+  // Return the remat'ed value in the given encoding.
+  Value getRematValue(Value value, Attribute encoding) {
+    return rematMapping[{value, encoding}];
```

Comment:
can you do this to avoid generating empty Value in case something went wrong and the key is not there:
```
auto it = rematMapping.find({value, encoding});
assert(it != rematMapping.end());
return it.second;
```